packages/toolkit/connector-kit/src/types.ts

```diff
@@ -181,6 +181,25 @@ export type GetSession = () => Promise<ConnectorSession>;
 
 export type SetSession = (storage: ConnectorSession) => Promise<void>;
 
+// https://github.com/colinhacks/zod#json-type
+const literalGuard = z.union([z.string(), z.number(), z.boolean(), z.null()]);
+
+type Literal = z.infer<typeof literalGuard>;
+
+export type JsonStorageValue = Literal | { [key: string]: JsonStorageValue } | JsonStorageValue[];
+
+export const jsonStorageValueGuard: z.ZodType<JsonStorageValue> = z.lazy(() =>
```

Comment:
This may cause performance issues in both dev and build time (This happended in recent days), the type calculation is too deep. I'll suggest using "unknown" or something similar.

Reply:
Use `ArbitraryObject`?